backend/src/bundles/stripe-events/helpers/subscription-email-generator.ts

```diff
@@ -0,0 +1,47 @@
+import { type SendMailOptions } from 'nodemailer';
+
+import { EmailPayload } from '../constants/email-payload.js';
+import { type SubscriptionDetails } from '../types/types';
+
+const loadAndReplaceTemplate = (variables: Record<string, string>): string => {
+    let htmlContent: string = EmailPayload.HTML;
+    for (const key of Object.keys(variables)) {
+        const regex = new RegExp(`{${key}}`, 'g');
+        htmlContent = htmlContent.replace(regex, variables[key]);
+    }
+
+    return htmlContent;
+};
+
+const generateSubscriptionEmailPayload = ({
+    email,
+    subject = 'Subscription',
+    name,
+    current_period_start,
+    current_period_end,
+}: SubscriptionDetails): SendMailOptions => {
+    const startDate = new Date(
+        current_period_start * 1000,
+    ).toLocaleDateString();
+    const endDate = new Date(current_period_end * 1000).toLocaleDateString();
```

Comment:
What is the magic number 1000?
maybe separate function formatDate for example to follow DRY?

Reply:
done